tests/ext/pynamodb/test_pynamodb.py

```diff
@@ -0,0 +1,73 @@
+import botocore.session
+import pytest
+from botocore.exceptions import ClientError
+from pynamodb.attributes import UnicodeAttribute
+from pynamodb.exceptions import VerboseClientError
+from pynamodb.models import Model
+
+from aws_xray_sdk.core import patch
+from aws_xray_sdk.core import xray_recorder
+from aws_xray_sdk.core.context import Context
+
+patch(('pynamodb',))
+
+
+@pytest.fixture(autouse=True)
+def construct_ctx():
+    """
+    Clean up context storage on each test run and begin a segment
+    so that later subsegment can be attached. After each test run
+    it cleans up context storage again.
+    """
+    xray_recorder.configure(service='test', sampling=False, context=Context())
+    xray_recorder.clear_trace_entities()
+    xray_recorder.begin_segment('name')
+    yield
+    xray_recorder.clear_trace_entities()
+
+
+def test_exception():
+    class SampleModel(Model):
+        class Meta:
+            region = 'us-west-2'
+            table_name = 'mytable'
+
+        sample_attribute = UnicodeAttribute(hash_key=True)
+
+    try:
+        SampleModel.describe_table()
+    except VerboseClientError:
+        pass
+
+    subsegments = xray_recorder.current_segment().subsegments
+    assert len(subsegments) == 1
+    subsegment = subsegments[0]
+    assert len(subsegment.subsegments) == 0
+    assert subsegment.error
+    assert subsegment.http['response']['status'] in [400, 403]
```

Comment:
I would suggest remove status code assertion. Any 4xx could be returned depend on setup of the developer's machine.

Reply:
done